Lesson16_Task01.py

```diff
@@ -0,0 +1,126 @@
+"""
+Home-task 13 Class writing Practice
+
+# write class for bank card.
+# Class should reflect card lifecycle, card operations etc.
+# use CVV, PIN, Name, Surname , end date, card_num as initial params
+# class should have in addition to common logic some class attributes, as minimum one classmethod and
+# as minimum  one staticmethod, two or more getters/setters
+# do not forget about block '''if __name__ == '__main__':'''
+"""
+
+import datetime
+
+
+class BankCard:
+    """
+    This class represents a bank card with various attributes and operations.
+    """
+
+    def __init__(self, card_num, cvv, pin, name, surname, end_date):
+        self.card_num = card_num
+        self.cvv = cvv
+        self.pin = pin
+        self.name = name
+        self.surname = surname
+        self.end_date = end_date
+        self.is_blocked = False
+        self.balance = 0.0
+
+    @classmethod
+    def generate_expiry_date(cls):
+        """
+        Generate an expiry date for the card.
+        """
+        today = datetime.date.today()
+        expiry_date = today.replace(year=today.year + 5)
+        return expiry_date
+
+    @staticmethod
+    def validate_card_number(card_num):
+        """
+        Validate the card number.
+        """
+        return len(card_num) == 16 and card_num.isdigit()
+
+    def block_card(self):
+        """
+        Block the card.
+        """
+        self.is_blocked = True
+
+    def deposit(self, amount):
+        """
+        Deposit money into the card's balance.
+        """
+        if not self.is_blocked and amount > 0:
+            self.balance += amount
+            return True
```

Comment:
I think deposit and withdraw should just return current balance. No sense to return true or false. Better in case of critical error raise exceptions. For example raise BaseException('Card is blocked') and then in main code catch them with try-except blocks